### DISCUSSION TITLE
How do I synchronize spawning a list of applications in groups?

### DISCUSSION THREAD

**Discussion by kkotyk** [qtile/qtile#5018] [Q&A]
What is the suggested way of spawning a list of applications in my groups and having those spawns be synchronized. Currently when I run this, I run into an issue where Qtile moves on to creating a new group before apps are finished spawning and they appear in the wrong groups.

```
    def initialize_groups(self):
        groups = []

        group_names = ["1","2","3","4","5","6","7","8","9","0"]
        group_layouts=["max", "max", "max", "max","max", "max", "max","max", "max", "max"]

        apps = [None] * len(group_layouts)
        apps[0] = "firefox"
        apps[1] = "subl"
        apps[2] = "kitty"
        apps[3] = "obsidian"
        apps[5] = "vesktop"

        for i in range(len(group_names)):
            groups.append(
                Group(
                name=group_names[i],
                layout=group_layouts[i].lower(),
                label=group_names[i],
                spawn=apps[i]
            ))

        return groups
```

**Comment by elParaguayo** [comment]
You could try adding match rules to the groups as well.

**Comment by kkotyk** [comment]
Would match rules make it so these apps only open in these windows? I basically want it so a set of programs are opened when I first log on, moved to specific groups and then have those groups moved to specific screens. Can match rules be used for this?
